container.go

```diff
@@ -34,6 +34,9 @@ var (
 	errReturnKind  = errors.New("constructor return type must be a pointer")
 	errArgKind     = errors.New("constructor arguments must be pointers")
 
+	// ErrInvokeOnce is returned if function is already invoked
+	ErrInvokeOnce = errors.New("function is already invoked")
```

Comment:
me and @glibsm were talking about whats the expected behavior on InvokeOnce. If InvokeOnce fail, should we silently continue or fail? There are some cases when we just want to continue. strongly typed error `dig.ErrInvokeOnce` can allow callers to decide whether they just want to log and continue or panic with an equals check.